content/en/tutorials/gitlab_ci_testcontainers/index.md

```diff
@@ -0,0 +1,431 @@
+---
+title: "End-to-End Testing in Gitlab CI with Testcontainers and LocalStack: Understanding Runners and Docker in Docker"
+linkTitle: "End-to-End Testing in Gitlab CI with Testcontainers and LocalStack: Understanding Runners and Docker in Docker"
+weight: 3
+description: >
+  In this tutorial, we'll walk through the process of setting up end-to-end testing for a backend application using Testcontainers and LocalStack within GitLab. We'll understand the types of GitLab Runners available for CI pipelines and how the concept of Docker-in-Docker plays a crucial role in this environment.
+type: tutorials
+teaser: "Learn how you can set up end-to-end testing using Testcontainers and LocalStack in GitLab CI. This tutorial covers setting up the .gitlab-ci.yml file, configuring GitLab runners, and setting up a local Docker runner. With this setup, you can ensure your application is thoroughly tested in a CI/CD environment."
+services:
+- s3
+- lambda
+- api-gw
+platform:
+- java
+deployment:
+- aws-java-sdk
+- aws-cli
+tags:
+- Java
+- AWS Java SDK
+- GitLab
+- GitLab Runners
+- Docker-in-Docker
+pro: true
+leadimage: "ls-gitlab-testcontainers.png"
+---
+
+## Introduction: Testcontainers & LocalStack
+
+Testcontainers is an open-source framework that provides lightweight APIs for bootstrapping local development and test dependencies
+with real services wrapped in Docker containers.
+Running tests with Testcontainers and LocalStack is crucial for AWS-powered applications because it ensures each test runs in a clean,
+isolated environment, providing consistency across all development and CI machines. LocalStack avoids AWS costs by emulating
+services locally, preventing exceeding AWS free tier limits, and eliminates reliance on potentially unstable external AWS services.
+This allows for the simulation of difficult-to-reproduce scenarios, edge cases, and enables testing of the
+entire application stack in an integrated manner. Testing with LocalStack and Testcontainers also integrates
+seamlessly with CI/CD pipelines like GitLab CI or GitHub Actions, allowing developers to run automated tests without requiring AWS credentials or services.
+
+## Prerequisites
+
+For this tutorial, you will need:
+
+- [LocalStack Pro](https://localstack.cloud/pricing/) to emulate the AWS services. If you don't have a subscription yet, you can just get a trial license for free.
```

Comment:
Please link to the installation doc instead of the pricing page and add `LOCALSTACK_AUTH_TOKEN` as well